Mathlib/Algebra/DirectSum/LinearMap.lean

```diff
@@ -18,11 +18,14 @@ domain and codomain.
 
 open Set BigOperators DirectSum
 
-variable {ι R M : Type*} [CommRing R] [AddCommGroup M] [Module R M]
-  {N : ι → Submodule R M} [DecidableEq ι] (h : IsInternal N)
-
 namespace LinearMap
 
+variable {ι R M : Type*} [CommRing R] [AddCommGroup M] [Module R M] {N : ι → Submodule R M}
+
+section IsInternal
```

Comment:
The name is not very clear (because `internal` has a mathematical meaning). Maybe `AuxiliaryConstruction` or just `Auxiliary`?

Reply:
This section has `variable (h : IsInternal N)`.

Reply:
Oh yeah. So the name is fine. I'd rather have `h` moved to the various lemmas to make them more readable, but it's not really in scope for this PR. So you can merge!